spi/core-spi/src/main/java/org/eclipse/dataspaceconnector/spi/query/ReflectionBasedQueryResolver.java

```diff
@@ -0,0 +1,85 @@
+/*
+ *  Copyright (c) 2020 - 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.spi.query;
+
+import org.eclipse.dataspaceconnector.common.reflection.ReflectionUtil;
+
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import static org.eclipse.dataspaceconnector.common.reflection.ReflectionUtil.propertyComparator;
+
+/**
+ * Applies query on a stream. Uses reflection to fetch object fields. Used in stores implementations.
+ *
+ * @param <T> type of the stream elements.
+ */
+public class ReflectionBasedQueryResolver<T> extends QueryResolver<T> {
```

Comment:
Can we move this into `:core:core-base`? it would then have to be passed into constructors in the `*Stores`